src/core/client/framework/lib/form/helpers.tsx

```diff
@@ -97,11 +97,15 @@ export const formatStringList = (v: string[] | null) => {
 
 export type FieldMeta = Pick<
   FieldRenderProps<any, HTMLElement>["meta"],
-  "touched" | "error" | "submitError"
+  "error" | "submitError" | "dirty" | "submitFailed"
 >;
 
-export const hasError = ({ touched, error, submitError }: FieldMeta) =>
-  touched && (error || submitError);
+export const hasError = ({
+  error,
+  submitError,
+  dirty,
+  submitFailed,
+}: FieldMeta) => (dirty || submitFailed) && (error || submitError);
```

Comment:
That's a nice improvement! Even better I think is leaving `dirty` out and only use `submitFailed && (error || submitError);`

Reply:
Yeah, this makes sense. Seconding that this is a nice fix @marcushaddon!

Reply:
this works perfecty!